src/mscorlib/corefx/System/Buffers/ArrayPool.cs

```diff
@@ -19,6 +19,16 @@ namespace System.Buffers
     /// </remarks>
     public abstract class ArrayPool<T>
     {
+        /// <summary>Per-type cached pool.</summary>
+        /// <remarks>
+        /// byte[] and char[] are the most commonly pooled array types. For these we use a special pool type
+        /// optimized for very fast access speeds, at the expense of more memory consumption.
+        /// The shared instance is created lazily on first access by the runtime.
+        /// </remarks>
+        private readonly static ArrayPool<T> _sharedInstance =
+            typeof(T) == typeof(byte) || typeof(T) == typeof(char) ? new TlsOverPerCoreLockedStacksArrayPool<T>() :
+            Create();
```

Comment:
The generic types are beforefieldinit by design, so it is not necessary to have a helper type for lazy initialization. 

I have factored this change into a separate PR with ArrayPool fixes: https://github.com/dotnet/coreclr/pull/8774